resources/aws_ec2.go

```diff
@@ -687,6 +689,16 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 		return false, fmt.Errorf("instance is nil")
 	}
 
+	// store useful send/recv values when we return
+	defer func() {
```

Comment:
Assuming instance is always set, I think this is actually a really clever way to do this! +1